src/grpc_server.rs

```diff
@@ -0,0 +1,431 @@
+
+        trace!("Transforming config string to hashmap");
+
+        let hashmap_config: HashMap<String, String> = match serde_json::from_str(&config) {
+            Ok(config) => {
+                trace!("JSON config parsed successfully for {}", deployment_uuid);
+                config
+            }
+            Err(e) => {
+                trace!("Failed to parse json config: {}", e);
+                return Err(Status::invalid_argument(format!(
+                    "Failed to parse json config: {}",
+                    e
+                )));
+            }
+        };
+
+        trace!(
+            "Creating app {} with config {:?}",
+            deployment_uuid,
+            hashmap_config
+        );
+
+        let app = Application {
+            application_id: deployment_uuid.clone(),
+            project_id: project_uuid,
+            image_name: format!("{}/{}", user_uuid, deployment_uuid),
+            image_tag: String::from("latest"),
+            env_variables: hashmap_config,
+        };
+
+        let message = match self.docker_engine.restart_application(&app).await {
+            Ok(_) => {
+                trace!("Applied config to {}", deployment_uuid);
+                format!("Applied config to application {}", deployment_uuid)
+            }
+            Err(e) => {
+                trace!(
+                    "Failed to apply config to application {}: {}",
+                    deployment_uuid,
+                    e
+                );
+                return Err(Status::internal(format!(
+                    "Failed to apply config to application {}: {}",
+                    deployment_uuid, e
+                )));
+            }
+        };
+
+        let response = ResponseMessage { message };
+        Ok(Response::new(response))
+    }
+}
+
+/// # Get App from db with uuid
+/// Get an `Application` from the database from its `uuid`, `project_uuid` and `user_uuid`.
+/// # Arguments
+/// - The borrowed `uuid` of the application to get.
+/// - The borrowed `project_uuid` of the application to get.
+/// - The borrowed `user_uuid` of the application to get.
+/// # Returns
+/// The application, wrapped in a Result.
+/// # Errors
+/// If the application is not found in the database, returns a `Status::not_found`.
+fn get_app(deployment_uuid: &str, project_uuid: &str, user_uuid: &str) -> Application {
+    Application {
+        application_id: deployment_uuid.to_string(),
+        project_id: project_uuid.to_string(),
+        image_name: format!("{}/{}", user_uuid, deployment_uuid),
+        image_tag: String::from("latest"),
+        ..Default::default()
+    }
+}
+
+/// # Start Server
+/// Start the gRPC server.
+/// # Arguments
+/// - The Docker engine to use to manage containers.
+/// - The database reference to use to get the deployments.
+/// # Returns
+/// Nothing.
+pub async fn start_server(docker_engine: DockerEngine) -> Result<(), Box<dyn std::error::Error>> {
+    let addr = "[::1]:50051".parse()?;
```

Comment:
Shouldn't this be `[::]` to allow all addresses ?

Also, perhaps the port shouldn't be hardcoded

Reply:
Pomegranate is containerized, so exposing a hardcoded port seems better than having it set in a .env, I believe ?